engine-src/Game/LambdaHack/Client/UI/TutorialHints.hs

```diff
@@ -0,0 +1,31 @@
+module Game.LambdaHack.Client.UI.TutorialHints
+  (TutorialHints (..)
+  , renderTutorialHints
+  ) where
+
+import Data.Text (Text)
+import Prelude
+
+data TutorialHints =
+    DamageOfDifferentKind
+    | NewFloorNewOpportunity
+    | CannotHarmYouInMelee
+    | CaughtProjectile
+    | HitsWithNoDirectDamage
+    deriving(Enum, Eq, Show)
+
+
+
+-- | Generate the standard textual representation according to
+--
+renderTutorialHints :: TutorialHints -> Text
+renderTutorialHints DamageOfDifferentKind =
```

Comment:
The syntax with `\case` would make it more succint:

```
= \case
NewFloor -> "text"
CaughtPro -> "awer"
```

Reply:
I agree, will do.